src/lib/types/Payloads.ts

```diff
@@ -77,6 +77,22 @@ export namespace Payloads {
     keys: string[];
   }
 
+  /**
+   * The payload for {@link Method.Ensure}
+   * @since 2.0.0
+   * @see {@link Payload}
+   * @see {@link Payload.Data}
+   */
+  export interface Each extends Payload {
+    /**
+     * The method this payload is for.
+     * @since 2.0.0
+     */
+    method: Method.Each;
+
+    hook: (value: any, key: string) => void;
```

Comment:
Same as the one above.

Reply:
> Tests need to be implemented as well.

@RealShadowNova: Would those go in JOSH or the JOSHProvider?